src/shared/particle_dynamics/general_dynamics/general_dynamics.h

```diff
@@ -61,6 +62,35 @@ namespace SPH
 		virtual ~BaseTimeStepInitialization(){};
 	};
 
+
+    class TimeStepInitializationKernel {
+    public:
+        template<class AccPriorType, class PosType, class GravityType>
+        static void update(size_t index_i, Real dt, AccPriorType acc_prior, PosType pos, GravityType&& gravity)
```

Comment:
First of all, thanks for the review!

The problem here is that when `static update` is used by `TimeStepInitialization` it passes a pointer, while `TimeStepInitializationKernel` passes a `sycl::accessor`. The latter cannot be changed, while for the pointer I can just pass the entire vector instead, therefore I think it's better to use a forwarding reference in this case.

Another option that I was evaluating lately is to use SYCL USM instead of buffers/accessors, in which case the kernel class would pass a pointer and your suggestion would be preferred, but that would probably be another PR I suppose.

Reply:
OK, context of use is clearer now. I agree that a forwarding reference would be the best option.